lib/cog/command/pipeline/executor.ex

```diff
@@ -532,14 +548,24 @@ defmodule Cog.Command.Pipeline.Executor do
         Helpers.send_error(error_message, state.request, state.mq_conn)
         fail_pipeline(state, :empty_output, error_message)
       _ ->
-        case command.execution do
-          "once" ->
+        case CommandCache.fetch(h) do
+          {:ok, command} ->
+            case command.execution do
+              "once" ->
+                {:next_state, :bind, %{state | current: h, remaining: t, input: [], output: [], context: new_output}, 0}
+              "multiple" ->
+                [oh|ot] = new_output
+                {:next_state, :bind, %{state | current: h, remaining: t, input: ot, output: [], context: oh}, 0}
+            end
+          {:not_found, "user:" <> _user_alias} ->
             {:next_state, :bind, %{state | current: h, remaining: t, input: [], output: [], context: new_output}, 0}
-          "multiple" ->
-            [oh|ot] = new_output
-            {:next_state, :bind, %{state | current: h, remaining: t, input: ot, output: [], context: oh}, 0}
+          {:not_found, "site:" <> _site_alias} ->
+            {:next_state, :bind, %{state | current: h, remaining: t, input: [], output: [], context: new_output}, 0}
+          {:not_found, command} ->
```

Comment:
We really need that refactor. I'm having a really hard time keeping this all in my head.